caluma/caluma_core/health_checks.py

```diff
@@ -0,0 +1,213 @@
+from django.conf import settings
+from django.db import connections
+from django.db.migrations.executor import MigrationExecutor
+from django.db.migrations.autodetector import MigrationAutodetector
+from django.db.migrations.state import ProjectState
+from django.apps.registry import apps
+from django.http import JsonResponse
+from django.core import management
+
+from caluma.caluma_logging.models import AccessLog
+from caluma.caluma_form import storage_clients
+
+import watchman
+from watchman import views
+from watchman.decorators import check
+from watchman import settings as watchman_settings
+
+from io import BytesIO, StringIO
+import json
+
+
+@check
+def _check_pending_migrations(db):
+    """
+    Check database for pending migrations.
+    Returns JSON mapping if no pending migrations, otherwise raises Exception.
+    @check decorator catches and handles exception.
+    """
+
+    # check for changes i.e. migrations that haven't been made
+    # makemigrations --check exists with exit status 1 if changes exist
+    try:
+        management.call_command("makemigrations", "--check", "--dry-run")
+    except SystemExit:
+        raise Exception("Migrations aren't up-to-date (makemigrations).")
+
+    # check for unapplied migrations
+    try:
+        out = StringIO()
+        management.call_command("migrate", "--plan", stdout=out)
+        plan = out.getvalue()
+        print(plan)
+        if not "No planned migration operations." in plan:
```

Comment:
I'd try to find another pattern or another way to check. This output is very likely translated and thus the check will stop working if someone runs it on a non-english machine (or possibly even when using a non-english browser during testing)

Reply:
Searching for pattern `[ ]` in output of `showmigrations --plan`.
For later versions: In Django 3.1 `migrate --plan` exits with a non-zero status when unapplied migrations are detected ([link](https://docs.djangoproject.com/en/3.1/ref/django-admin/#migrate)).